Core/src/IfcGeometryConverter/SolidModelConverter.h

```
@@ -174,9 +182,69 @@ namespace OpenInfraPlatform
+					std::cout << "vec_cross_section_positions decreased by one" << std::endl;
+					vec_cross_section_positions.pop_back();
+				}
+
+                 //TO DO: check lenght conversions on the ProfileDistanceExpression L[2:?] CrossSectionPostitions
+				double length_in_meter = UnitConvert()->getLengthInMeterFactor();
+
+
+				 //Give directrix to Curve converter
+		    	 std::vector<carve::geom::vector<3> > segment_start_points;
+	     		 std::vector<carve::geom::vector<3> > basis_curve_points;
+				 curveConverter->convertIfcCurve(directrix.lock(), basis_curve_points, segment_start_points);
+
+				 std::shared_ptr<carve::input::PolyhedronData> pipe_data(new carve::input::PolyhedronData());
+				 itemData->closed_polyhedrons.push_back(pipe_data);
+				 //std::vector<carve::geom::vector<3> > inner_shape_points;  //TO DO: find out if i need inner_shape_points for the CrossSections
+
+				 int num_curve_points = basis_curve_points.size();
+				 carve::math::Matrix matrix_sweep;
+
+				 // Less than two points is a point
+				 if (num_curve_points < 2)
+				 {
+					 std::cout << "IfcSectionedSolidHoritzontal: num curve points < 2" << std::endl;
+					 return;
+				 }
+				 
+				 //Get coordinates from the ProfileConverter for the ProfileDef
+				 for (int i = 0; i <= vec_cross_sections.size(); ++i)
+			     {       
+					     //activate profile converter for each Cross Section 
+						 std::shared_ptr<ProfileConverterT<IfcEntityTypesT>> profile_converter = profileCache->getProfileConverter(vec_cross_sections[i]);
+						 profile_converter->simplifyPaths();
+						 //Save coordinates in paths
+						 const std::vector<std::vector<carve::geom::vector<2>>>& paths = profile_converter->getCoordinates();
+		         }
+  
+			     // check if paths has been filled with the coordinates of the ProfileDef. if it is empty -> return;
+				 if (paths.size() == 0)
+				 {
+					 std::cout << "Profile converter could not find coordinates" << std::endl;
+					 return;
+				 }

```

These should be within the for loop above. Plus, it should throw an exception.

---

resolved in commit ccd8948